File: competitive_programming/codeforces/07_typescript/116a_tram.ts
function tram(stops: number[][]): number {
  let current = 0;
  let capacity = 0;
  for (const [leaving, entering] of stops) {
    current = current - leaving + entering;
    capacity = Math.max(capacity, current);
  }
  return capacity;
}

if (tram([[0, 3], [2, 5], [4, 2], [4, 0]]) !== 6) throw new Error("tram failed");
console.log("116A tram ok");

File: competitive_programming/codeforces/09_rust/116a_tram.rs
fn tram(stops: &[(i32, i32)]) -> i32 {
    let mut current = 0;
    let mut capacity = 0;
    for &(leaving, entering) in stops {
        current += entering - leaving;
        capacity = capacity.max(current);
    }
    capacity
}

fn main() {
    assert_eq!(tram(&[(0, 3), (2, 5), (4, 2), (4, 0)]), 6);
    println!("116A tram ok");
}
